File: app/src/main/java/net/chris/demo/guardian/DetailActivity.java
package net.chris.demo.guardian;

import android.content.res.Configuration;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

import static net.chris.demo.guardian.database.Columns.WEB_URL;

public class DetailActivity extends AppCompatActivity {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.fragment_detail);
        if (fragment.getArguments() == null) {
            fragment.setArguments(new Bundle());
        }
        fragment.getArguments().putString(WEB_URL, getIntent().getStringExtra(WEB_URL));
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
    }

}

File: app/src/main/kotlin/net/chris/demo/guardian/DetailActivity.kt
package net.chris.demo.guardian

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import net.chris.demo.guardian.database.Columns

class DetailActivity : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)
        val fragment = supportFragmentManager.findFragmentById(R.id.fragment_detail)
        if (fragment.arguments == null) {
            fragment.arguments = Bundle()
        }
        fragment.arguments.putString(Columns.WEB_URL, intent.getStringExtra(Columns.WEB_URL))
    }

}
